deploy/common/k8sbased.sh

```diff
@@ -322,6 +322,17 @@ function launch_central {
         helm lint "$unzip_dir/chart" -n stackrox
         helm lint "$unzip_dir/chart" -n stackrox "${helm_args[@]}"
       fi
+
+      # set custom central version if the CENTRAL_CHART_VERSION env is set
+      # this has to happen after linting since helm lint does not recognize the --version flag
+      if [[ ! -z ${CENTRAL_CHART_VERSION} ]]; then
+        helm_args+=(
+          --version="${CENTRAL_CHART_VERSION}"
+        )
```

Comment:
It does not look like this actually works.
See https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/stackrox_stackrox/5795/pull-ci-stackrox-stackrox-master-gke-version-compatibility-tests/1661704732579729408/artifacts/gke-version-compatibility-tests/stackrox-e2e/build-log.txt

```
Calling: helm upgrade --install -n stackrox stackrox-central-services /go/src/github.com/stackrox/stackrox/deploy/k8s/central-deploy//chart                                -f /go/src/github.com/stackrox/stackrox/deploy/k8s/central-deploy//values-public.yaml -f /go/src/github.com/stackrox/stackrox/deploy/k8s/central-deploy//values-private.yaml --set-string imagePullSecrets.useExisting=stackrox;stackrox-scanner -f /go/src/github.com/stackrox/stackrox/deploy/common/ci-values.yaml --set system.enablePodSecurityPolicies=true --version=74.3.0
Release "stackrox-central-services" does not exist. Installing it now.
W0525 14:07:00.816038  183001 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0525 14:07:00.877244  183001 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0525 14:07:00.939956  183001 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0525 14:07:04.771271  183001 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0525 14:07:04.771408  183001 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0525 14:07:04.775127  183001 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: stackrox-central-services
LAST DEPLOYED: Thu May 25 14:06:59 2023
NAMESPACE: stackrox
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
StackRox 4.0.x-424-gf9a2748d4c has been installed.
```

Notice that `--version=74.3.0` is provided to `helm` command, but `StackRox 4.0.x-424-gf9a2748d4c has been installed.` comes out in the log.

This is like shooting in darkness in attempt to hit the target: we provide a path to the local chart but specify a different version. We hope there's **some Helm repository** with such version but we don't tell which repository it is. To me it's no surprise `helm` installs the chart from the provided directory and does not try to search repositories. It is a bit surprising that `helm` command doesn't fail due to the version mismatch though.

Before going the path of also overriding chart directory with repo indicator, I'd warn that we have multiple Helm repos and we don't necessarily always take charts from <https://github.com/stackrox/helm-charts/tree/main/opensource>. Also, it's not necessary that `helm` has all repositories already configured and have expected names (remember: this script runs both in CI and on developer machines).
Let's what is the right place to override the chart version (disclaimer: I don't have an answer).

Reply:
After searching it seems to me that helm deployments all happen in `deploy/common`k8sbased.sh` so I think it is the correct place for overriding the helm chart and version.For these compatibility tests in particular shouldn't the opensource repo be sufficient?

But I can see that if we have an override we might want to get charts from other repos too so how about a `HELM_CHART_OVERRIDE_NAME` and `HELM_CHART_OVERRIDE_URL` which set the name of the repo and URL to it? I could then set those to `stackrox-oss` and `https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource` for the compatibility tests.